src/main/java/org/assertj/core/internal/Iterables.java

```diff
@@ -1108,6 +1109,60 @@ public void assertContainsExactly(AssertionInfo info, Iterable<?> actual, Object
       throw failures.failure(info, elementsShouldSatisfy(actual, unsatisfiedRequirements, info));
   }
 
+  /**
+   * Verifies that all given consumers can be satisfied by elements in the iterable under test with
+   * an element at most satisfying one consumer. No order requirement.
+   *
+   * @param info contains information about the assertion.
+   * @param actual the given {@code Iterable}.
+   * @param consumers the consumers that are expected to be satisfied by the elements of the given {@code Iterable}.
+   * @throws NullPointerException if the given consumers array or var args comsumers is {@code null}.
+   * @throws AssertionError if the given {@code Iterable} is {@code null}.
+   * @throws AssertionError if any {@code Consumer} in the consumers cannot be satisfied by elements in the given {@code Iterable}.
+   */
+  @SafeVarargs
+  public final <E> void assertSatisfy(AssertionInfo info, Iterable<? extends E> actual, Consumer<? super E>... consumers) {
+    assertNotNull(info, actual);
+    requireNonNull(consumers, "The Consumer<? super E>... expressing the assertions consumers must not be null");
+    for(Consumer<? super E> consumer: consumers)
+      requireNonNull(consumer, "The element in consumers must not be null");
+
+    List<E>[] satisfiedElementsLists = new ArrayList[consumers.length];
+    for (int i = 0; i < consumers.length; i++) {
+      Consumer<? super E> consumer = consumers[i];
+      satisfiedElementsLists[i] = stream(actual).filter(element -> {
+        try {
+          consumer.accept(element);
+        } catch (AssertionError ex) {
+          return false;
+        }
+        return true;
+      }).collect(toList());
+    }
+    if (!isSatisfied(satisfiedElementsLists, 0))
+      throw failures.failure(info, shouldSatisfy(actual));
+  }
+
```

Comment:
There is already a method ```byPassingAssertions```, which takes care of the try/catch-block around ```consumer.accept```. 
Should we extract some methods to make the intend of the main method clearer?

```suggestion
    List<?>[] satisfiedElementsLists = Arrays.stream(consumers)
                                             .map(listFilteredBySatisfiedElements(actual))
                                             .toArray(List[]::new);
    
    if (!isSatisfied(satisfiedElementsLists))
      throw failures.failure(info, shouldSatisfy(actual));
  }

  private <E> Function<? super Consumer<? super E>, ?> listFilteredBySatisfiedElements(Iterable<? extends E> actual) {
    return consumer -> stream(actual).filter(byPassingAssertions(consumer)).collect(toList());
  }

  private static <E> boolean isSatisfied(List<E>[] satisfiedElementsLists) {
    return isSatisfied(satisfiedElementsLists, 0);
  }


```

Reply:
The stream processing part is great, but I'm not sure it is worth having another overload for `isSatisfied`.

Reply:
Would it be better to change the type of `satisfiedElementsLists` from `List<?>[]` to `List<E>[]` or `List[]`?

Reply:
I understand your concern about overloading ```isSatisfied```. I often use such overloading to have a dedicated entry point for recursive methods. But I agree, the overloaded method might distract more than it helps. Feel free to remove it. 

I think the best type of ```satisfiedElementsLists``` is ```List<E>[]```. To avoid compiler warnings, it could be written as: 
```
    List<E>[] satisfiedElementsLists = Arrays.stream(consumers)
                                             .map(listFilteredBySatisfiedElements(actual))
                                             .<List<E>>toArray(List[]::new);
```
```suggestion
    List<E>[] satisfiedElementsLists = Arrays.stream(consumers)
                                             .map(listFilteredBySatisfiedElements(actual))
                                             .<List<E>>toArray(List[]::new);
    
    if (!isSatisfied(satisfiedElementsLists, 0))
      throw failures.failure(info, shouldSatisfy(actual));
  }
  private <E> Function<? super Consumer<? super E>, List<E>> listFilteredBySatisfiedElements(Iterable<? extends E> actual) {
    return consumer -> stream(actual).filter(byPassingAssertions(consumer)).collect(toList());
  }

```

Reply:
maybe it would be better to static import `stream`, just as we did in method `Iterables.assertAllSatisfy`